File: C++/Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int readPoint = 0, firstPoint = 0, nextPoint = 1;
        while (firstPoint < n) {
            A[readPoint] = A[firstPoint];
            readPoint++;
            if (nextPoint < n) {
                while (A[firstPoint] == A[nextPoint]) {
                    nextPoint++;
                    if (nextPoint >= n) {
                        break;
                    }
                }
                firstPoint = nextPoint;
                nextPoint++;
            } else {
                break;
            }
        }
        return readPoint;
    }
};

File: Java/Remove Duplicates from Sorted Array.java
public class Solution {
	public int removeDuplicates(int[] nums) {
		// version 1
		int l = nums.length;
		int i, j;
		for (i = 0; i < l - 1; i++)
			if (nums[i] == nums[i + 1])
				break;
		for (j = i + 2; j < l; j++)
			if (nums[j] != nums[i])
				nums[++i] = nums[j];
		return ++i;
		
		// version 2
		int l = nums.length;
		int j = 0;
		for (int i = 1; i < l; i++)
			if (nums[i] != nums[j])
				nums[++j] = nums[i];
		return j + 1;
	}
}